File: 70_climbing_stairs/dp.py
class Solution:
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        """
        Complexity: O(n)
        Space: O(n)
        """
        ans = [0]*(n+1)
        if n >= 1:
            ans[1] += 1
            if n >= 2:
                ans[2] += 2
            else:
                return 1
        else:
            return 0
        for i in range(3, n+1):
            ans[i] += (ans[i-1] + ans[i-2])
        return ans[-1]

File: 070_climbing_stairs/dp.cpp
class Solution {
public:
    int climbStairs(int n) {
        std::vector<int>dp(n+1);
        if(n >= 1){
            dp[1] += 1;
            if(n >= 2) dp[2] += 2;
            else return 1;
        }
        else return 0;
        for(int i = 3; i < n+1; ++i){
            dp[i] += dp[i-1] + dp[i-2];
        }
        return dp[n];
    }
};
